### PR TITLE
feat(auth): add SignInState

### PR DESCRIPTION
**Pull request by div5yesh**
- [x] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*

*Description of changes:*
- Add SignInState as AuthN substate to handle other auth scenarios such as custom auth and MFA.

*How did you test these changes?*
(Please add a line here how the changes were tested)

- [ ] Added Unit Tests
- [x] Tested on frontend (Add Screenshots)
- [ ] Successful logs (Attach them to the PR)

*Documentation update required?*
- [x] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by tylerjroach — APPROVED** [review]

**Review by gpanshu — DISMISSED** [review]

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/SignInState.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+            oldState.srpSignInState?.let { srpSignInResolver.resolve(it, event) }?.let {
+                builder.srpSignInState = it.newState
+                actions += it.actions
+            }
+
+            return StateResolution(builder.build(), actions)
+        }
+
+        private fun resolveSRPSignInEvent(
```
Comment (tylerjroach) [inline aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/SignInState.kt:47]: should this say `resolveSignInEvent`

Comment (div5yesh) [reply aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/SignInState.kt]: Yes.

**Review by gpanshu — APPROVED** [review]

**Review by sktimalsina — COMMENTED** [review]
wonder how Projext.xml file in .idea still show up

**Comment by div5yesh** [comment]
It is tracked in Git to standardize coding style in IDE.

**Review by tylerjroach — APPROVED** [review]

**Pull request merged**